src/main/java/de/tum/in/www1/artemis/service/compass/CompassCalculationEngine.java

```diff
@@ -262,44 +310,125 @@ public JsonObject getStatistics() {
         return jsonObject;
     }
 
-    private boolean hasConflict(int elementId) {
-        boolean conflict = false;
-        Assessment assessment = this.assessmentIndex.getAssessmentsMap().get(elementId);
-        if (assessment == null) {
-            return false;
-        }
-        for (List<Score> scores: this.assessmentIndex.getAssessmentsMap().get(elementId).getContextScoreList().values()) {
-            double uniqueScore = -1;
-            for (Score score: scores) {
-                if (uniqueScore != -1 && uniqueScore != score.getPoints()) {
-                    conflict = true;
+
+    private void buildAssessment(long submissionId, List<ModelElementAssessment> modelingAssessment) {
+        UMLModel model = modelIndex.getModel(submissionId);
+        Map<String, Score> scoreList = createScoreList(modelingAssessment, model);
+        this.addNewManualAssessment(scoreList, model);
+        modelSelector.removeModelWaitingForAssessment(model.getModelID());
+    }
+
+
+    private Map<String, Score> createScoreList(List<ModelElementAssessment> modelingAssessment, UMLModel model) {
```

Comment:
This code is rather hard to understand, can you add some comments and describe what you actually want to achieve? I guess you search for a modelElement with a specific ID.
I suggest you use some of the Java 8 Lambda features, e.g. stream() and filter(...) to replace the for loops.

Reply:
I just adapted this method to handle the new ModelElementAssessment but did not change the actual logic since im not quiet sure why he is doing all this and if for example the check for existence of each jsonElementId in the UMLModel is really necessary. I just added a javadoc to this method with my understanding of what this method does.

Reply:
Since our planned refactoring for the Feedback in ModelingAssessments would replace this method i would leave it how it is until then.

Reply:
ok